src/IW/Effects/Download.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
+
+module IW.Effects.Download where
+
+import Network.HTTP.Client (Manager, Request (..), Response (..), httpLbs)
+import Network.HTTP.Types (Status (..))
+
+import IW.App (App (..), grab, Has, WithError)   
+import IW.Core.Url (Url (..)) 
+
+
+class Monad m => MonadDownload m where
+    downloadFile :: Url -> m ByteString
+
+instance MonadDownload App where 
+    downloadFile = downloadFileImpl
+
+type WithDownload env m = (MonadIO m, MonadReader env m, WithError m, Has Manager env)
+
+downloadFileImpl :: WithDownload env m => Url -> m ByteString
+downloadFileImpl Url{..} = do
+    man <- grab @Manager
+    let req = fromString $ toString unUrl
+    response <- liftIO $ httpLbs req man
+    pure $ case statusCode $ responseStatus response of
+        200       -> toStrict $ responseBody response
+        otherCode -> undefined
```

Comment:
We should use `throwError` function in here instead.

Reply:
@vrom911 Yep will add that. I guess for now we can just have a simple text error, but in a separate issue I propose to make types for download errors. Here #110

Reply:
Oh, I see! Thanks for pointing that out :slightly_smiling_face: